src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list.isEmpty()) {
+            return;
+        }
+
+        for (Object object : list) {
+            T nodeElement = (T) object;
+            Node<T> newNode = new Node<>(last, nodeElement, null);
+            if (last == null) {
+                first = newNode;
+            } else {
+                last.next = newNode;
+            }
+            last = newNode;
+        }
+        size += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T set(T item, int index) {
+        Node<T> nodeValueByIndex = getNode(index);
+        T oldItem = nodeValueByIndex.item;
+        nodeValueByIndex.item = item;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> nodeValue = first; nodeValue != null; nodeValue = nodeValue.next) {
+            if ((object == null) && (nodeValue.item == null)
+                    || (object != null) && (object.equals(nodeValue.item))) {
+                unlink(nodeValue);
+                return true;
+            }
+        }
+
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void linkLast(T value) {
+        final Node<T> last = this.last;
+        final Node<T> newNode = new Node<>(last, value, null);
+        this.last = newNode;
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkBefore(T value, Node<T> succ) {
+        final Node<T> pred = succ.prev;
```

Comment:
I believe the variable names "succ" and "pred" may be changed to more meaningful